rust_src/src/minibuf.rs

```diff
@@ -65,4 +69,51 @@ pub fn minibuffer_prompt() -> LispObject {
     unsafe { Fcopy_sequence(minibuf_prompt) }
 }
 
+/// Return the buffer position of the end of the minibuffer prompt.
+/// Return (point-min) if current buffer is not a minibuffer.
+#[lisp_fn]
+pub fn minibuffer_prompt_end() -> EmacsInt {
+    let beg = ThreadState::current_buffer().beg() as EmacsInt;
+    if memq(current_buffer(), unsafe { Vminibuffer_list }).is_nil() {
+        return beg;
+    }
+
+    let end = field_end(Some(beg), false, None);
+    let buffer_end = ThreadState::current_buffer().zv as EmacsInt;
+    if end == buffer_end && get_char_property(beg, Qfield, LispObject::constant_nil()).is_nil() {
+        beg
+    } else {
+        end
+    }
+}
+
```

Comment:
Do it makes sense to collapse these two into an internal function? The only difference is the true/false for adding text properties. I know the C did not do it. Should we?

Reply:
Hmmm. I'm not sure if this is worth an additional function.